src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ThemeService.cs

```diff
@@ -139,47 +160,206 @@ public async Task<Either<ActionResult, List<ThemeViewModel>>> GetThemes()
                 );
         }
 
-        public async Task<Either<ActionResult, Unit>> DeleteTheme(Guid themeId)
+        public async Task<Either<ActionResult, Unit>> DeleteTheme(
+            Guid themeId,
+            CancellationToken cancellationToken = default)
         {
             return await _userService.CheckCanManageAllTaxonomy()
-                .OnSuccess(
-                    () => _persistenceHelper.CheckEntityExists<Theme>(
-                        themeId,
-                        q => q.Include(t => t.Topics)
-                    )
-                )
+                .OnSuccess(() => _persistenceHelper.CheckEntityExists<Theme>(themeId, q => q.Include(t => t.Publications)))
                 .OnSuccessDo(CheckCanDeleteTheme)
                 .OnSuccessVoid(async theme =>
                 {
-                    var topicIds = theme.Topics.Select(topic => topic.Id).ToList();
-                    await topicIds.ForEachAsync(_topicService.DeleteTopic);
+                    var publicationIdsToDelete = theme.Publications
+                        .Select(p => p.Id)
+                        .ToList();
+
+                    await DeleteMethodologiesForPublications(publicationIdsToDelete)
+                       .OnSuccess(() => DeleteReleasesForPublications(publicationIdsToDelete))
+                       .OnSuccess(() => DeletePublications(publicationIdsToDelete));
```

Comment:
I lose the value of `theme` in the following chain with those changes, any ideas?
![image](https://github.com/user-attachments/assets/0002f498-f4fa-406c-8a55-7f10dba1790c)

Reply:
Ah my bad - I think you just need the `return` and everything else remains the same. 

The way I think about `OnSuccessDo` is that if it doesn't return an ActionResult, it will pass through the result of the previous OnSuccess.

So in this case, the theme used in scopes starting lines 169, 170, and 180 are all using the theme fetched via `CheckEntityExists`, because of the OnSuccessDos.

When we changed line 170 to OnSuccess, it was now passing the result of that scope to line 180, so the result of the `return await DeleteMethodologies...` chain which if successful is a `Unit`.